ic-canister/ic-canister-macros/src/derive.rs

```diff
@@ -28,6 +28,35 @@ impl Default for TraitNameAttr {
     }
 }
 
+struct Interval {
+    mins: u64,
+}
+
+impl Parse for Interval {
+    fn parse(input: ParseStream) -> syn::Result<Self> {
+        input.parse::<Ident>()?;
+        input.parse::<Token![=]>()?;
+
+        if let Ok(lit) = input.parse::<syn::LitInt>() {
+            let value = lit.base10_parse::<u64>()?;
+            return Ok(Interval { mins: value });
+        }
+
+        let ident = input.parse::<syn::Ident>()?;
+        let interval = match ident.to_string().as_str() {
+            "min" => Interval { mins: 1 },
+            "hour" => Interval { mins: 60 },
+            "day" => Interval { mins: 60 * 24 },
+            "week" => Interval { mins: 60 * 24 * 7 },
+            "month" => Interval { mins: 60 * 24 * 30 },
```

Comment:
Also update anywhere in the docs / examples where this is used
```suggestion
            "hourly" => Interval { mins: 60 },
            "daily" => Interval { mins: 60 * 24 },
            "weekly" => Interval { mins: 60 * 24 * 7 },
            "monthly" => Interval { mins: 60 * 24 * 30 },
```